pkg/sensors/accelerator/device/device.go

```diff
@@ -0,0 +1,141 @@
+/*
+Copyright 2024.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+	http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package device
+
+import (
+	"errors"
+	"time"
+
+	"golang.org/x/exp/maps"
+	"k8s.io/klog/v2"
+)
+
+var (
```

Comment:
Per previous comment re: Accelerators, I'd prefer encapsulating this into a `typ DeviceRegistry struct`

Reply:
If it **really** needs to be global because it's being used to register plugins via `init` functions - not 100% sure I can think of an easier way to deal with this right now....

Then let's add some scaffolding like so:

```
var globalRegistry *DeviceRegistry
var once sync.Once

// DeviceRegistry gets the default device registry instance
func DeviceRegistry() *DeviceRegistry {
    once.Do(func() {
        globalRegistry = NewDeviceRegistry()
    })
    return globalRegistry
}

// SetDeviceRegistry replaces the global registry instance
// NOTE: All plugins will need to be manually registered
// after this function is called.
func SetDeviceRegistry(registry *DeviceRegistry) {
    globalRegistry = registry
}
```

That would allow us to replace the registry for testing purposes.

Reply:
I couldn't easily get rid of the global aspect as there are functions invoked from different packages throughout kepler to get the stats. 
So either I created a registry at startup and then passed a reference to it to every place that needed it in Kepler (passing it to all the stats collection functions at init time so they have a reference) or add the scaffolding as you suggested throughout - I opted for the scaffolding to avoid too many changes in the existing code base